### PR TITLE
Update sbt-web dependencies

### PR DESCRIPTION
**Pull request by marcospereira**
Updates sbt-web (and plugins) dependencies, since we already made some releases already. This is related to issue #7045.

### CODE REVIEW

File: framework/project/Dependencies.scala [hunk 1]
```diff
@@ -190,8 +190,8 @@ object Dependencies {
 
       sbtDep("com.typesafe.sbt" % "sbt-native-packager" % BuildInfo.sbtNativePackagerVersion),
 
-      sbtDep("com.typesafe.sbt" % "sbt-web" % "1.3.0"),
-      sbtDep("com.typesafe.sbt" % "sbt-js-engine" % "1.1.3")
+      sbtDep("com.typesafe.sbt" % "sbt-web" % "1.4.0"),
+      sbtDep("com.typesafe.sbt" % "sbt-js-engine" % "1.1.4")
```
Comment (benmccann) [inline framework/project/Dependencies.scala]: We still need a newer version. This one doesn't have any of the latest changes. See https://github.com/sbt/sbt-js-engine/issues/52

File: framework/project/Dependencies.scala [hunk 2]
```diff
@@ -31,17 +31,17 @@ object Dependencies {
     "com.fasterxml.jackson.core" % "jackson-databind",
     "com.fasterxml.jackson.datatype" % "jackson-datatype-jdk8",
     "com.fasterxml.jackson.datatype" % "jackson-datatype-jsr310"
-  ).map(_ % "2.8.8")
+  ).map(_ % "2.8.8.1")
```
Comment (benmccann) [inline framework/project/Dependencies.scala]: This is not a valid version for most of these libraries, so this change is breaking the build

File: framework/project/Dependencies.scala (hunk 1 above)
Comment (mkurz) [inline framework/project/Dependencies.scala]: Maybe we also should cut a new release for `sbt-web`? There are some [new commits](https://github.com/sbt/sbt-web/commits/master) in the master branch.

File: framework/project/Dependencies.scala (hunk 2 above)
Comment (marcospereira) [inline framework/project/Dependencies.scala]: Removed.

File: framework/project/Dependencies.scala (hunk 1 above)
Comment (marcospereira) [inline framework/project/Dependencies.scala]: We can update again after releasing both sbt-js-engine and sbt-web.

I just want to keep the pace with these dependencies so we don't have bigger updates (and maybe some breakage) later.

**Comment by benmccann** [comment]
It looks like the upgrade to `play-file-watch` is causing an issue:

>java.nio.file.NoSuchFileException: /tmp/sbt_9c64325/dev-mode/target/jnotify/META-INF/MANIFEST.MF

@wsargent tried upgrading just that one library in https://github.com/playframework/playframework/pull/7277 and is getting the same error in his build

**Comment by marcospereira** [comment]
@benmccann removed play-file-watch update here, so that it can be updated in #7277 (including investigating why it is failing).

**Pull request merged**